client/src/components/EditAssessment/EditAssessment.tsx

```diff
@@ -15,16 +15,31 @@ const EditAssessment = ({
   onSave,
   assessment,
 }: EditAssessmentProps) => {
-  const [name, setName] = useState(assessment.name);
-  const [date, setDate] = useState(assessment.date);
-  const [weight, setWeight] = useState(assessment.weight);
+  const [name, setName] = useState<string>(assessment.name);
+  const [date, setDate] = useState<Date>(assessment.date);
+  const [weight, setWeight] = useState<number>(assessment.weight);
   const [location, setLocation] = useState("");
   const [modality, setModality] = useState("");
   const [notes, setNotes] = useState("Late submission policy applied.");
 
   const handleWeightChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    if (isNaN(Number(e.target.value.trim()))) return;
-    setWeight(e.target.value.trim());
+    if (isNaN(Number(e.target.value.trim()))) {
+      return;
+    }
+    if (e.target.value.trim() === "") {
+      setWeight(0);
+      return;
+    }
+    setWeight(Number(e.target.value.trim()));
+  };
+
+  const jsxInputFormat = (date: Date): string => {
+    const year = date.getFullYear();
+    const month = `${date.getMonth() + 1}`.padStart(2, "0");
+    const day = `${date.getDate()}`.padStart(2, "0");
+    const hours = `${date.getHours()}`.padStart(2, "0");
+    const minutes = `${date.getMinutes()}`.padStart(2, "0");
+    return `${year}-${month}-${day}T${hours}:${minutes}`;
```

Comment:
Can use `toISOString()` instead.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString

Reply:
I did that at first, but it would display in UTC time instead of local